areas/room.cpp

```diff
@@ -660,8 +660,9 @@ void displayRoom(Player* player, const BaseRoom* room, int magicShowHidden) {
     for(Exit* ext : room->exits) {
         wallOfFire = ext->isWall("wall-of-fire");
         wallOfThorns = ext->isWall("wall-of-thorns");
+        hasBeenUsedBy = ext->hasBeenUsedBy(player);
```

Comment:
Been a while since I've looked at showExit, but might be worth moving the check in there, potentially after some other less expensive checks.  Should also only compute this if it's discoverable.